apps/api/src/app/shared/services/cache/cache.service.ts

```diff
@@ -0,0 +1,72 @@
+import Redis from 'ioredis';
+import { getRedisPrefix } from '@novu/shared';
+
+interface ICacheService {
+  set(key: string, value: string, options?: CachingConfig);
+  get(key: string);
+  del(key: string);
+  reset();
+}
+
+export type CachingConfig = {
+  ttl?: number;
+};
+
+export class CacheService {
+  private readonly client: Redis;
+  private readonly DEFAULT_TTL = 60;
+
+  constructor() {
+    this.client = new Redis(Number(process.env.REDIS_PORT), process.env.REDIS_HOST, {
```

Comment:
Should we bring the REDIS_PORT and REDIS_PASSWORD from the config file rather than the process directly?

Reply:
I refactored it a bit in the following branches however I still fetch from the env so the user could modify it from it (as for the self-hosted etc) as all our configuration params, do you think we could suggest adding those secrets from another place? @p-fernandez

Reply:
Oh no, I guess I suggested config because in case there is a name refactor for this env variables you have a single place where they come from. Functionality wise indeed has zero problems. 👍🏻

Reply:
@p-fernandez 
In order to make it more robust we changed it to CAHCE_HOST, CAHCE_PORT, etc ..
once again sorry for the confusion but I added it on the following PR.